sdk/metric/periodic_reader.go

```diff
@@ -214,11 +220,12 @@ func (r *periodicReader) aggregation(kind InstrumentKind) aggregation.Aggregatio
 // the SDK and exports it with r's exporter.
 func (r *periodicReader) collectAndExport(ctx context.Context) error {
 	// TODO (#3047): Use a sync.Pool or persistent pointer instead of allocating rm every Collect.
-	rm := metricdata.ResourceMetrics{}
-	err := r.Collect(ctx, &rm)
+	rm := r.rmPool.Get().(*metricdata.ResourceMetrics)
+	err := r.Collect(ctx, rm)
 	if err == nil {
-		err = r.export(ctx, rm)
+		err = r.export(ctx, *rm)
```

Comment:
Probably not for this PR, but should the `Export` interface be updated to accept a `*ResourceMetrics` instead of `ResourceMetrics? Given Go is a pass-by-value language, passing the already allocated pointer would reduce the call-stack size.

Reply:
https://github.com/open-telemetry/opentelemetry-go/issues/3852